index.html

```diff
@@ -273,6 +283,48 @@ <h2>Network requests</h2>
       <a>HTTP-network fetch</a> algorithm.
       </p>
 
+      <p>
+      Regardless of which fetch algorithm and which underlying application and
+      transport protocols are used, servicing a <a>network request</a> consists
+      of the following <dfn data-lt="phase">phases</dfn>:
+      </p>
+
+      <ol>
+        <li>
+          <dfn>DNS resolution</dfn>: The user agent uses the Domain Name System
+          [[RFC1034]] to resolve a domain name into an IP address of a
+          <a>server</a> can that service HTTP requests to that domain.
+        </li>
+
+        <li>
+          <dfn>Secure connection establishment</dfn>: The user agent opens a

```

Comment:
Should we distinguish TCP from TLS here? Would also be good to properly define what happen with those phases in protocols which don't include them (TCP-Fast-Open + TLS/1.3 0-RTT, QUIC).

Reply:
I tried to make this not too granular, exactly because of the different protocols and fast paths that there are.  I didn't want this to be "a taxonomy of connection methods as of the time of this writing"; instead I wanted "in the broadest possible strokes, here's what happens when the user agent makes a request over the network".  So I tried to merge phases together as much as I could.

These three seemed like the minimum — I couldn't glom any more of them together.  `dns` has to be separate because it's treated differently, both for downgraded reports and for the handling of `include_subdomains`.  `application` has to be separate because it's the only mandatory phase.  It didn't seem necessary to split apart the `connection` phase — at least here in NEL — because it wouldn't have any impact on whether we're allowed to collect a report, or on what information that report could contain.

Reply:
OK. I can see use cases for distinguishing between them on the collector side (e.g. distinguishing between "host is down" to "cert issues"). At the same time, not sure how important that distinction is, so fine with leaving it out for now. I also agree that in a world where everything is QUIC, that distinction may not make much sense.

Reply:
There are still all of the `tcp.*` and `tls.*` error codes, so the collector will still be able to see a fine-grained description of what error occurred.  It's just that all of those error codes would have the same value (`connection`) for their `phase` field.

Reply:
Oh, OK. missed that